Updated editor spin slider to have better behaviour and adjusted control's size_flags_stretch_ratio value range

EricEzaM opened the pull request:
Fixes #37786

A few changes

1. Updated range of stretch ratio and added `or_greater` to allow for values > max if required. Max of 20 was selected because it's still small enough to perform minor adjustments at lower numbers, and it is conceivable that someone would want a control to take 5% of the width/height reasonably often. (or should it be lower?)
2. Pressing 'control' makes the slider round to the nearest value... but previously it was making the value jump since the cached mouse drag distance went from being *1 to *10 instantly (see below). This no longer happens.
3. If you kept dragging below min or above max, you would have to drag your mouse alllllll the way back in order to get into range. Now, if you are at the minimum value, a change in direction of the mouse is all you need to go back to being in range, and seeing the value update.
4. Added documentation of the shift/control shortcuts... not sure if the class documentation is the right place for it, but there isn't really a "Editor tips and tricks" page afaik. Let me know if I should remove that from this PR.

**BEFORE**
You can see when I press control the value jumps by difference * 10, and back when I let go. When I am at zero for a while it is because I am scrolling far to the 'negative' and I have to keep scrolling in the positive to get back.
![nbNfDpH9Vb](https://user-images.githubusercontent.com/41730826/81073875-9a1a4500-8f2b-11ea-947b-8ce48d98fc91.gif)
**AFTER**
Neither of those issues here...
![dODSeL35YV](https://user-images.githubusercontent.com/41730826/81073860-94bcfa80-8f2b-11ea-8f4f-4f7e37368d0b.gif)

bojidar-bg commented on doc/classes/Control.xml:
```diff
@@ -865,7 +865,7 @@
 		</member>
 		<member name="size_flags_stretch_ratio" type="float" setter="set_stretch_ratio" getter="get_stretch_ratio" default="1.0">
-			If the node and at least one of its neighbours uses the [constant SIZE_EXPAND] size flag, the parent [Container] will let it take more or less space depending on this property. If this node has a stretch ratio of 2 and its neighbour a ratio of 1, this node will take two thirds of the available space.
+			If the node and at least one of its neighbours uses the [constant SIZE_EXPAND] size flag, the parent [Container] will let it take more or less space depending on this property. If this node has a stretch ratio of 2 and its neighbour a ratio of 1, this node will take two thirds of the available space. If using the editor slider with the mouse, hold the Control key to round values to the nearest ...
```
Pretty sure this should not be in the class documentation.

It might still be nice to have it in a tooltip or in some editor documentation.

EricEzaM replied:
I agree... I just put it there because I wanted it documented, but I was quite sure it was not the right place. I'll see if I can put it into a tool tip or something... Didn't look for that last night when I did it.

EricEzaM replied:
See update... removed doco and added tooltip.

EricEzaM committed:
Updated editor spin slider to have better behaviour and adjusted control's size_flags_stretch_ratio value range

EricEzaM commented:
Removed update to documentation.

Added tooltip for sliders to show this hint if the grabber (white circle) is visible, meaning it does not show on SpinSliders where the up/down arrows are shown. Is this the best place for it? Is it worth having a tooltip like this? Or should there just be a 'Editor Tips + Tricks' docs page or something where small stuff like this is noted.
![godot windows tools 32_cY7umI9S4X](https://user-images.githubusercontent.com/41730826/81128713-48060d80-8f85-11ea-9a1b-d970ad20ad7f.png)

akien-mga commented on editor/editor_spin_slider.cpp:
```diff
@@ -36,6 +36,9 @@
 
 String EditorSpinSlider::get_tooltip(const Point2 &p_pos) const {
+	if (grabber->is_visible()) {
+		return rtos(get_value()) + "\nHold Ctrl to round to integers. Hold Shift for more precise changes.";
```
```suggestion
		return rtos(get_value()) + "\n" + TTR("Hold Ctrl to round to integers. Hold Shift for more precise changes.");
```
Makes it translatable.

EricEzaM replied:
Thanks!

akien-mga commented on editor/editor_spin_slider.cpp:
```diff
@@ -111,7 +114,21 @@ void EditorSpinSlider::_gui_input(const Ref<InputEvent> &p_event) {
 			}
 
 			if (grabbing_spinner) {
+				// Don't make the user scroll all the way back to 'in range' if they went off the end
```
We're about to settle on a comment style close to this, but ending sentences with a dot. Could you add the final dot here and on line 126?

akien-mga commented:
Thanks!

akien-mga commented:
Cherry-picked for 3.2.2.
